### ISSUE TITLE
SA1137/SA1316 false positive when using primary constructors

### ISSUE THREAD

**Issue by Emik03** [DotNetAnalyzers/StyleCopAnalyzers#3663]
## Problem
The following code block causes false warnings. (`LangVersion` set to `preview`)
```cs
class A(int b)
{
    public int B => b;
}
```
![image](https://github.com/DotNetAnalyzers/StyleCopAnalyzers/assets/14614115/a1a76357-5ec9-43aa-89a1-f10eef6983a0)
![image](https://github.com/DotNetAnalyzers/StyleCopAnalyzers/assets/14614115/b348a166-93f6-4f95-b5af-ae9a4f462cae)

## How this is a problem
It is suggesting to format the code like so:
```cs
class A(int B)
{
public int B => b;
}
```

When using primary constructors in non-record types, [they're considered parameters/fields](https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/proposals/primary-constructors), and therefore use `camelCase`.

The lack of indentation is clearly unintended since the same thing isn't raised for record types, and it goes against the rule of contents in blocks having indentation in the first place.

**Comment by bjornhellander** [comment]
I tried with Visual Studio 17.6.2 and StyleCop Analyzers 1.2.0-beta.435, but can not reproduce this.
Can you provide some info about your setup?

**Comment by Emik03** [comment]
Operating System: Linux (popOS)
IDE: Rider 2023.2 EAP 2
.NET SDK: `8.0.100-preview.4.23260.5`

```xml
<ItemGroup>
    <PackageReference Include="StyleCop.Analyzers" Version="1.2.0-beta.435" PrivateAssets="all"/>
</ItemGroup>
```

I also [reference `Emik.Morsels`](https://github.com/Emik03/Emik.Morsels), which is essentially a toolchain/build tool. I don't see why it would conflict with the analyzer, but it is part of my environment.

<details> 
<summary>Detailed information about my IDE setup</summary>
JetBrains Rider 2023.2 EAP 2
Build #RD-232.6095.13, built on May 18, 2023
Licensed to JetBrains Rider EAP user: Emik Emik
Expiration date: June 17, 2023
Runtime version: 17.0.7+7-b953.1 amd64
VM: OpenJDK 64-Bit Server VM by JetBrains s.r.o.
Linux 6.2.6-76060206-generic
.NET Core v7.0.4 x64
GC: G1 Young Generation, G1 Old Generation
Memory: 8000M
Cores: 12
Registry:
    actionSystem.playback.typecommand.delay=4
    ide.intellij.laf.enable.animation=true
    actionSystem.playback.delay=4
    documentation.show.toolbar=true
    ide.experimental.ui=true
    ide.balloon.shadow.size=0
    ide.new.project.model.index.case.sensitivity=true
    vcs.empty.toolwindow.show=false
    database.show.search.tab=false

Non-Bundled Plugins:
    manjaro.epb (1.7)
    com.4lex4.intellij.solarized (2.4.0)
    com.berzanmikaili.intellij.monokai_pro (2.0.2c)
    com.pandocker (0.7.3)
    com.thvardhan.gradianto (5.1)
    org.rust.lang (0.4.194.5382-231)
    org.toml.lang (232.6095.10)
    com.mallowigi.idea (21.0.0)
    com.jetbrains.darkPurpleTheme (1.3)
    monokai-pro (1.10)
    com.vermouthx.idea (1.14.3)
    com.jetbrains.rider.android (232.6095.13)
    com.intellij.ideolog (203.0.30.0)
    com.godwin.json.parser (1.4.1)
    com.tylerthrailkill.intellij.solarized (3.0.0)
    String Manipulation (9.7.1)
    CMD Support (1.0.5)
    com.arcticicestudio.nord.jetbrains (0.13.0)
    com.samdark.intellij-visual-studio-code-dark-plus (2.7)
    org.vlang (0.0.1-beta.4)
    com.github.holgerbrandl.pasteimages/ (1.2.4)
    mobi.hsz.idea.gitignore (4.5.0)

Current Desktop: pop:GNOME
</details>

**Comment by bjornhellander** [comment]
I actually got very similar StyleCop warnings before upgrading to VS 17.6, i.e. when using an older compiler version, but then I of couse got proper compiler errors as well, not recognizing the new syntax. Based on how this analyzer is implemented, getting this warning in your case looks impossible, so I am suspecting some kind of compiler version mismatch. Does this happen when building from the command line as well, or only inside Rider?

**Comment by Emik03** [comment]
Interesting, this only happens inside Rider.

![image](https://github.com/DotNetAnalyzers/StyleCopAnalyzers/assets/14614115/16395475-aa5e-437c-972b-6a81f72b9e07)

This is weird, because in theory it shouldn't have a problem since I use the newest toolchain, but I suppose it is because Rider runs on .NET Core v7.0.4 (see detailed information above)

![image](https://github.com/DotNetAnalyzers/StyleCopAnalyzers/assets/14614115/fab6e519-de50-417c-bf50-39edc40332f0)

**Comment by Emik03** [comment]
Just wanted to chime in one last time to mention that this indeed has been fixed as of version 2023.2 EAP 5 with the latest toolchain preview.

![image](https://github.com/DotNetAnalyzers/StyleCopAnalyzers/assets/14614115/17a9ce0d-355d-4c3f-bbe7-438dcaa6dc7c)
